README.md

```diff
@@ -10,103 +10,150 @@
+## Install
 
-### distribution index versions / updates
+First, install the following dependencies via your favorite package manager:
 
-The **distribution index** changes over time, kind of like a git repository. [Since we don't yet have commits](https://github.com/ipfs/notes/issues/23), we will just do a poor-man's versioning for the index itself. We will write all version hashes to a file `versions` in this repository.
+* yarn
+* hugo
+* npm
+* jq
+* ipfs
+* git (obviously)
 
-A site like `dist.ipfs.io` or `ipfs.io/dist` would just serve the _latest_ version of the index.
-
-## Install
+Then install the javsacript dependencies with yarn:
 
 ```sh
-$ git clone https://github.com/ipfs/distributions
-$ cd distributions
 # Install javascript dependencies
-$ yarn
-# Install dependency for make files
-$ npm install -g jq
+> yarn
 ```
 
-This project uses a makefile + scripts to build all the things.
+Finally, run `make` to build/download the existing distribution over IPFS. It will download everything published at dist.ipfs.io and then build anything missing.
 
 ```sh
-$ make
+> make
 ```
 
-should do everything.
+## Usage
 
+### Adding a version
 
-## Usage
+Run:
+
+```sh
+> ./dist.sh add-version <dist> <version>
+```
+
+This will add the version to `dists/<dist>/versions`, set it as the current version in `dists/<dist>/current`, and build it.
+
+### Adding a new (go) distribution
+
+Run:
 
-Each `<dist>` has a directory in the root of this repo. inside it there is a `Makefile` and other necessary scripts. Running
+```sh
+> ./dist.sh new-go-dist <dist> <git-repo>
+```
+
+And follow the prompts.
+
+### Publishing 
 
+In the root of the repository, run:
+
+```sh
+> make publish
 ```
-make
+
+This will build (or download) anything that hasn't been built and build, compile the index in `releases`, and add releases to ipfs. Save the hash it spits out, that's the new hash for `dists.ipfs.io`.
+
+Next, you should probably compare this with the current `dists.ipfs.io` to make sure nothing is amiss:
+
+```sh
+> ipfs object diff /ipns/dist.ipfs.io /ipfs/<NEW_HASH>
 ```
 
-Should:
-- figure out what the latest released version is (from github tags)
-- figure out what versions are missing from the index
-- construct the missing `<dist>/<version>` directories
+If all looks well, pin the hash using pinbot (#ipfs-pinbot on Freenode, ask
+someone if you don't have permission to do so).
+
+Finally,
+
+1. Commit your changes and make a PR. Specifically, the changes to `dists/<dist>/versions` and `dists/<dist>/current`.
+2. File an issue on [ipfs/infrastructure](https://github.com/ipfs/infrastructure) with the hash you got from `make publish` and a link to the PR.
+
+If you have permission, you can just merge the PR, update the DNS, and then immediately, close the issue on ipfs/infrastructure. Ping someone on IRC.
```

Comment:
Add something about rememebering to pin the new hash with the pinbot, because I just forgot that step

Reply:
Look up.

Reply:
indeed